### ISSUE TITLE
[placeholder] Change click behavior in files

### ISSUE THREAD

**Issue by marcoambrosini** [nextcloud/server#41787]
Single click: select
Double click: open

On touch devices the behaviour stays the same as current

**Comment by marcoambrosini** [comment]
@SystemKeeper @nickvergessen could you please elaborate the sad face? :)

**Comment by nickvergessen** [comment]
> could you please elaborate on the sad face? :)

It's confused, not sad, but I used it as "uncomfortable".

> It's especially useful to do multi-select operations when the target area is the whole row, as opposed to just the checkbox.

I get that, the thing is I mostly never use those operations. In 99% of the cases I just want to see the file in big/editor and not start a mass operation.

My biggest issue is (similar to the right click), that this makes the files list behave much different compared to all the other lists we have. Contacts, User list for admins, App list, etc.

Does this also apply to the grid layout?

**Comment by marcoambrosini** [comment]
https://baymard.com/blog/users-double-click-online

**Comment by jancborchardt** [comment]
> https://baymard.com/blog/users-double-click-online

From the article (which is from 2013):

> August 2019 update: This user behavior is increasing and even more prevalent now than when this article was first published. However, this increase is mainly because of the mobile aspect “The user simply thinks their tab wasn’t registered” whenever the mobile site response is just slightly slow (as addressed further down the article).

So it seems that nowadays (further 5 years on from the update) it is more about performance and feedback, rather than expecting double click.

**Comment by skjnldsv** [comment]
@marcoambrosini is this still wanted ?
- desktop: click = select / double-click = open 
- touch devices: tap = open?

**Comment by p1gp1g** [comment]
I've been using Nextcloud for more than 5 years now, and I have never get used to the UX to open/select/see details. Most of the time, I wish to see the details of a file and end up downloading it, I wish to select multiple files and end up downloading one, etc.

I think the best behavior would be:
- With a mouse:
    - Single click = open details (and select?)
    - Single click on the file/folder icon = open (not sure for this one, maybe a download icon/open icon would be better)
    - Maj + Single click (icon or not) = select (this is a common behavior)
    - If at least one file is selected, single click (icon or not) = select (This is very important IMO, and could be done even before the other things are implemented)
    - Double click = open

- With touch screen (this is currently how the Nextcloud app works on Android,  but not the web app):
    - Tap = open
    - Long click = select
    - If at least one file is selected, tap = select

**Comment by nextcloud-command** [comment]
This issue has been automatically marked as stale because it has not had recent activity and seems to be missing some essential information. It will be closed if no further activity occurs. Thank you for your contributions.

**Comment by marcoambrosini** [comment]
@jancborchardt should we close?

**Comment by jancborchardt** [comment]
@marcoambrosini yes, I'd agree on closing this. I share Joas' opinion above, and we haven't seen a mass of feedback about the current state feeling wrong for lots of people.

As time goes on, single tap will just become more and more common.

**Comment by p1gp1g** [comment]
Do you mind reconsider it, this issue is not only about the single click, isn't it ?
There are:
- the question to match behavior on touch screen with the mobile app
- the point for MAJ + click to select (and see details ?)
- the point of a single click during a selection, which is always used to (un)select an item,

Should I open 3 issues for them then ?

**Comment by p1gp1g** [comment]
I've filled the issues, except for the behavior with a touch screen, I don't use the web app on mobile. If you care about it, I let you open it

**Comment by jancborchardt** [comment]
@p1gp1g we just closed this issue though, so if there is any follow-up we should keep it here and not open 3 new ones in place of one we said is not planned? :)

**Comment by jancborchardt** [comment]
@p1gp1g thanks for opening the 3 separate issues, makes it much easier to tackle them. :)
